driver-core/src/test/unit/com/mongodb/client/model/TestWindowedComputations.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model;
+
+import com.mongodb.lang.Nullable;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.junit.jupiter.api.Test;
+
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static com.mongodb.client.model.Windows.documents;
+import static com.mongodb.client.model.Windows.range;
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+final class TestWindowedComputations {
+    private static final String NO_EXPRESSION = "{}";
+    private static final String PATH = "newField";
+    private static final Map.Entry<Integer, BsonValue> INT_EXPR = new AbstractMap.SimpleImmutableEntry<>(1, new BsonInt32(1));
+    private static final Map.Entry<String, BsonValue> STR_EXPR =
+            new AbstractMap.SimpleImmutableEntry<>("$fieldToRead", new BsonString("$fieldToRead"));
+    private static final Window POSITION_BASED_WINDOW = documents(1, 2);
+    private static final Window RANGE_BASED_WINDOW = range(1, 2);
+
+    @Test
+    void simpleWindowFunctions() {
```

Comment:
Question: Whats the rationale / motivation for these tests? Do we do similar for the other builders or do they just have output style tests as in AggregatesSpecification?

Reply:
Yes, this class contains unit tests which are conceptually similar to those in `AggregatesSpecification`: we know what the builders are supposed to build, and we check that the results are correct or that building fails for illegal arguments.

Since the signatures of many builder methods differ only in the method name, the unit tests for them were combined to omit repeating the same test logic.

Reply:
Ack